BWEnv/src/controller.cc

```diff
@@ -576,8 +573,13 @@ BWAPI::TilePosition Controller::getTilePositionFromWalkTiles(int x, int y) {
 int Controller::getAttackFrames(int unitID) {
   int attackFrames = BWAPI::Broodwar->getLatencyFrames();
   int unitType = BWAPI::Broodwar->getUnit(unitID)->getType().getID();
+  
   // From
   // https://docs.google.com/spreadsheets/d/1bsvPvFil-kpvEUfSG74U3E5PLSTC02JxSkiR8QdLMuw/edit#gid=0
+  // Photon Cannons aren't included in this chart but may also have a nonzero value.
+  // This value is primarily relevant for knowing how many frames a unit needs to be allowed to
+  // peform its attack animation before it can receive another command.
+  // Receiving a command before that period will cancel the attack.
```

Comment:
Thanks for elaborating, but I'm still a bit unsure about this since you don't handle photon cannons in the code below?

Reply:
I don't know what the value is for Photon Cannons. Never did the experiment myself. I just strongly suspect that it's non-zero.

Reply:
Hehe ok then a TODO would not hurt ;)

Reply:
Done!